src/lake/Lake/Build/Module.lean

```diff
@@ -213,9 +211,10 @@ def Module.recBuildLeanBcToO (self : Module) : IndexBuildM (BuildJob FilePath) :
 def Module.oFacetConfig : ModuleFacetConfig oFacet :=
   -- TODO: add option to switch between `.bc` and `.o` builds.
   mkFacetJobConfig (fun module =>
-    match module.backend with
-    | .C => module.recBuildLeanCToO
-    | .LLVM => module.recBuildLeanBcToO
+    if Lean.Internal.hasLLVMBackend () then
```

Comment:
I think which module is used here (in the `o` facet) should be determined by the backend configuration.  The generate everything based on `hasLLVMBackend` just had to do with the module build itself. In addition, it would be ideal to have `bc.o` and `c.o` facets to generate the o files for either output and just determine which one to fetch here based on the the `backend` setting.

Reply:
Done, I've added new facets to reflect this change.